Framework/DataHandling/src/MaskDetectors.cpp

```diff
@@ -191,10 +239,92 @@ void MaskDetectors::exec() {
   /*
   This rebuild request call, gives the workspace the opportunity to rebuild the
   nearest neighbours map
-  and therfore pick up any detectors newly masked with this algorithm.
+  and therefore pick up any detectors newly masked with this algorithm.
   */
   WS->rebuildNearestNeighbours();
 }
+/* Verifies input ranges are defined and returns these ranges if they are.
+*
+* @return tuple containing min/max ranges provided to algorithm
+*        (from 0 to max histogram range) and Boolean value,
+*        containing true if the ranges are
+*        constrained (or defined in other words)
+*        and false if they are not.
+*/
+std::tuple<size_t, size_t, bool>
+MaskDetectors::getRanges(const MatrixWorkspace_sptr &targWS) {
+  int endIndex = getProperty("EndWorkspaceIndex");
+  int startIndex = getProperty("StartWorkspaceIndex");
+  size_t max_ind = targWS->getNumberHistograms() - 1;
+
+  if (endIndex == EMPTY_INT() && startIndex == 0) {
+    return std::tuple<size_t, size_t, bool>(0, max_ind, false);
+  } else {
+    if (startIndex < 0) {
+      startIndex = 0;
+    }
+    size_t startIndex_l = static_cast<size_t>(startIndex);
+    size_t endIndex_l = static_cast<size_t>(endIndex);
+
+    if (endIndex == EMPTY_INT()) {
+      endIndex_l = max_ind;
+    }
+    if (endIndex_l > max_ind) {
+      endIndex_l = max_ind;
+    }
+    if (startIndex_l > endIndex_l) {
+      startIndex_l = endIndex_l;
+    }
```

Comment:
The shorthand for this would be
```
startIndex_1 = std::min(startIndex_1, endIndex_1);
```
but the `if` statement here is perhaps more readable.